### PR TITLE
[Audit Service] DB Config Priority and Doc Update

### PR DESCRIPTION
**Pull request by sthanikan2000**
## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update
- [x] Refactoring (no functional changes)
- [x] Performance improvement
- [ ] Other (please describe):

## Changes Made

feat(database): Enhance database configuration handling and add support for in-memory SQLite

- Updated NewDatabaseConfig to prioritize database type and path settings.
- Default to in-memory SQLite when no configuration is provided.
- Added tests for various database configurations including DB_PATH alone and DB_TYPE=sqlite.
- Improved logging for database configuration selection.

docs(api): Add comprehensive API documentation

- Created API.md for detailed endpoint descriptions, request/response formats, and examples.
- Included success and error response formats for audit log operations.
- Documented system endpoints for health checks and version information.

docs(architecture): Document project architecture and structure

- Added ARCHITECTURE.md outlining the directory structure and responsibilities of each layer.
- Explained design patterns used in the project, including Repository and DTO patterns.

docs(database): Provide detailed database configuration guidelines

- Created DATABASE_CONFIGURATION.md to explain database modes and configuration priorities.
- Included examples for in-memory, file-based SQLite, and PostgreSQL setups.
- Documented environment variables and their usage for database configuration.

## Testing

- [x] I have tested this change locally
- [x] I have added unit tests for new functionality
- [x] I have tested edge cases
- [x] All existing tests pass

## Checklist

- [x] My code follows the project's style guidelines
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: audit-service/docs/DATABASE_CONFIGURATION.md [hunk 1]
```diff
@@ -0,0 +1,365 @@
+# Database Configuration
```
Comment (ginaxu1) [inline audit-service/docs/DATABASE_CONFIGURATION.md]: Add a section in `DATABASE_CONFIGURATION.md` explaining the fallback behavior for unknown `DB_TYPE` values 

In audit-service, the code defaults to file-based SQLite with default path `./data/audit.db`

**Review by ginaxu1 — DISMISSED** [review]

File: audit-service/docs/DATABASE_CONFIGURATION.md (hunk 1 above)
Comment (ginaxu1) [reply audit-service/docs/DATABASE_CONFIGURATION.md]: see: https://github.com/OpenDIF/opendif-core/pull/420/files#diff-d69e43398682c5c64282bd419f43baf258564daf83f7b1e3b8593c2c8ba0215f
